docs/guides/deployment/fly_io.rst

```diff
@@ -0,0 +1,174 @@
+.. _ref_guide_deployment_fly_io:
+
+=========
+On Fly.io
+=========
+
+In this guide we show how to deploy EdgeDB to Fly.io using a Fly.io
+PostgreSQL cluster as the backend.
+
+
+Prerequisites
+=============
+
+* Fly.io account
+* ``flyctl`` CLI (`install <flyctl-install_>`_)
+
+.. _flyctl-install: https://fly.io/docs/getting-started/installing-flyctl/
+
+
+Provision a Fly.io app for EdgeDB
+=================================
+
+Every Fly.io app must have a globally unique name, including service VMs like
+Postgres and EdgeDB.  Here we assume the name for the EdgeDB app is
+"myorg-edgedb", which you would need to replace with a name of your choosing.
+
+.. code-block:: bash
+
+    $ EDB_APP=myorg-edgedb
+    $ flyctl apps create --name $EDB_APP
+    New app created: myorg-edgedb
+
+Now, let's secure the pending EdgeDB instance with a strong password:
+
+.. code-block:: bash
+
+    $ read -s EDGEDB_PASSWORD
+    <enter-password>
+    $ flyctl secrets set EDGEDB_PASSWORD="$EDGEDB_PASSWORD" -a $EDB_APP
+    Secrets are staged for the first deployment
+
+There are a couple more environment variables we need to set:
+
+.. code-block:: bash
+
+    $ flyctl secrets set \
+        EDGEDB_SERVER_BACKEND_DSN_ENV=DATABASE_URL \
```

Comment:
Is `EDGEDB_SERVER_BACKEND_DSN_ENV` the right spelling? Isn't it `EDGEDB_SERVER_BACKEND_DSN` without the `_ENV`?

Reply:
It's not a typo. `_ENV` specifies the name of _another_ environment variable where the actual value is, like `_FILE` specifies a file name.